### PR TITLE
Support BranchProbabilityInfo in update_analyze_test_checks.py

### PR DESCRIPTION
**Pull request by MatzeB**
- Change `BranchProbabilityPrinterPass` output to match expectations of `update_analyze_test_checks.py`.
- Add `Branch Probability Analysis` to list of supported analyses.
- Process `llvm/test/Analysis/BranchProbabilityInfo/basic.ll` with `update_analyze_test_checks.py` as proof of concept. Leaving the other tests unchanged to reduce the amount of churn.

### CODE REVIEW

File: llvm/utils/UpdateTestChecks/common.py [hunk 1]
```diff
@@ -773,12 +781,7 @@ def process_run_line(
             )
             if "analysis" in m.groupdict():
                 analysis = m.group("analysis")
-                supported_analyses = {
-                    "cost model analysis",
-                    "scalar evolution analysis",
-                    "loop access analysis",
-                }
-                if analysis.lower() not in supported_analyses:
```
Comment (WenleiHe) [inline llvm/utils/UpdateTestChecks/common.py:781]: I don't know why we need to normalize to lower case, but the change seems fine.

**Review by WenleiHe — APPROVED** [review]
lgtm, thanks.

**Review by artagnon — COMMENTED** [review]
Please add a test in `test/tools/UpdateTestChecks`.

File: llvm/utils/UpdateTestChecks/common.py (hunk 1 above)
Comment (MatzeB) [reply llvm/utils/UpdateTestChecks/common.py:781]: I don't think this normalization added anything. It did make it harder to search for the list of approved analyses though, so I used the original spelling from the code now.

**Pull request merged**